blog/static/img/2022-01-03-pcie_screamer.md

```diff
@@ -0,0 +1,301 @@
+```bash
+wget https://www.openssl.org/source/openssl-1.1.1o.tar.gz
+cd openssl-1.1.1o
+./config
+make
+make test
+sudo make install
+find / -name libssl.so.1.1
+ln -s /usr/local/lib64/libssl.so.1.1  /usr/lib64/libssl.so.1.1
+ln -s /usr/local/lib64/libssl.so.1.1  /usr/lib/libssl.so.1.1
+find / -name libcrypto.so.1.1
+ln -s /home/ubuntu/openssl-1.1.1o/libcrypto.so.1.1    /usr/lib64/libcrypto.so.1.1
+ln -s /home/ubuntu/openssl-1.1.1o/libcrypto.so.1.1     /usr/lib/libcrypto.so.1.1
+```
+After building the openssl library, a command was issued to test the communication
+of the PCIScreamer board with the pcileech application:
+
+```bash
+mgabryelski@maciej-HP:~/PCIScreamer$ sudo ./pcileech probe -device fpga -v
+
+DEVICE: FPGA: ERROR: Unable to load FTD3XX.dll [0,v0.0,0000]
+PCILEECH: Failed to connect to the device.
+mgabryelski@maciej-HP:~/PCIScreamer$ 
+```
+The FTDI for USB 3.0 chip is used to connect the PCIScreamer board with a PC, 
+the driver is not embedded in the Linux kernel. So I compiled the driver
+for the FTDI chip from the sources in this Github repository: [FTDI FT50x driver](https://github.com/ufrisk/pcileech/releases/tag/v4.15) The compilation was successful, 
+and after it was finished we have such files in working directory:
+
+```bash
+mgabryelski@maciej-HP:~/GIT1/pcie_screamer/drivers/ft60x$ ls
+51-ft60x.rules  ft60x.ko   ft60x.mod.c  ft60x.o     ftd3xx.h    Makefile       Module.symvers
+ft60x.c         ft60x.mod  ft60x.mod.o  ftconfig.c  loopback.c  modules.order
+mgabryelski@maciej-HP:
+```
+We try to load kernel module with FTDI driver issuing command:
+
+```bash
+mgabryelski@maciej-HP:~/PCIScreamer$ sudo insmod ft60x.ko
+````
+
+After that the kernel module is loaded:
+
+```bash
+mgabryelski@maciej-HP:~/PCIScreamer$ lsmod
+Module                  Size  Used by
+ft60x                  20480  0
+algif_skcipher         16384  0
+af_alg                 32768  1 algif_skcipher
+binfmt_misc            24576  1
+```
+Now checking `pcileech` application we have;
+```bash
+gabryelski@maciej-HP:~/PCIScreamer$ sudo ./pcileech probe -device fpga -v
+
+DEVICE: FPGA: ERROR: Unable to load FTD3XX.dll [0,v0.0,0000]
+PCILEECH: Failed to connect to the device.
+mgabryelski@maciej-HP:~/PCIScreamer$ 
+```
+As we can see, the problem still exists. I tried also building `pcileech` application 
+from sources following this tutorial [PCILeech on Linux](https://github.com/ufrisk/pcileech/releases/tag/v4.15).
+Unfortunately, all attempts to communicate the `pcileech` application with the 
+PCIScreamer FPGA board ended in failure, therefore, more advanced  tests were 
+not feasible.
+
+## Problems with the stability of the PCIE Screamer board
+
+PCIScreamer in revision R01 (which we have) is known for its stability problems
+and with some motherboards it does not work at all.
+
+The incorrect power delivered to the FPGA could be the result of the faulty PCIe
+low-profile riser card used to connect the PCIeScreamer to the motherboard.
```

Comment:
So the FPGA is not properly powered and this is the cause of problem with USB communication? Is this something confirmed or a suspiction only? Shouldn't the device be completely invisible on USB in case of power failure? Yet it is visible, just `pcileech` tool doesn't work.